app/post.go

```diff
@@ -810,7 +810,13 @@ func (a *App) GetPostsPage(options model.GetPostsOptions) (*model.PostList, *mod
 		}
 	}
 
-	if appErr := a.filterInaccessiblePosts(postList); appErr != nil {
+	// we use the slower route if not a collapsed threads request;
+	// Even though we know the parent and child subqueries are ordered within themselves, they are not added to the list in an ordered fashion with respect to each other, and there is not a 1:1 correspondence between Order and Posts (root posts are not added to Order).
+	filterOptions := filterPostOptions{}
```

Comment:
It looks to me, that `rootPosts` is a superset of `parentsPosts`. And since we add `order` from ~~parentsPosts~~ rootPosts only, then posts must be sorted. So, I think in this case we can set `assumeSortedCreatedAt` to `true`.

Although _another topic_, I don't see why add parentsPosts at all, as the rootPosts(superSet) is already being added.